packages/kbn-generate-csv/src/generate_csv.ts

```diff
@@ -532,4 +571,108 @@ export class CsvGenerator {
     const byteSize = pitId ? Buffer.byteLength(pitId, 'utf-8') : 0;
     return pitId?.substring(0, 12) + `[${byteSize} bytes]`;
   }
+
+  private async generateDataFromESQL(): Promise<TaskRunResult> {
+    const [settings, searchSource] = await Promise.all([
+      getExportSettings(
+        this.clients.uiSettings,
+        this.config,
+        this.job.browserTimezone,
+        this.logger
+      ),
+      this.dependencies.searchSourceStart.create(this.job.searchSource),
+    ]);
+
+    let reportingError: undefined | ReportingError;
+    const { maxSizeBytes, bom, escapeFormulaValues } = settings;
+    const builder = new MaxSizeStringBuilder(this.stream, byteSizeValueToNumber(maxSizeBytes), bom);
+    const warnings: string[] = [];
+
+    const searchBody = searchSource.getSearchRequestBody();
+    const searchParams: IKibanaSearchRequest<ESQLSearchParams> = {
+      params: {
+        // @ts-ignore
+        query: searchSource.getField('query')!.esql,
+        filter: searchBody.query,
+        locale: 'en', // TODO: get locale from request
+      },
+    };
+
+    this.logger.debug(`Search params: ${JSON.stringify(searchParams)}`);
+
+    try {
+      const { rawResponse } = await lastValueFrom(
+        this.clients.data.search(searchParams, {
+          strategy: ESQL_SEARCH_STRATEGY,
+          // TODO: not supported by esql search strategy
+          // transport: {
+          //   maxRetries: 0, // retrying reporting jobs is handled in the task manager scheduling logic
+          //   requestTimeout: settings.scroll.duration,
```

Comment:
These should be supported. I think they would end up just being passed to the low-level ES client. Reports often cover large time ranges, e.g. monthly/quarterly reports. A slow response from ES should be expected here, and retrying here could be costly.

Reply:
Yes, definitely agree. I added it to the todo list in the issue